component/discovery/kubernetes_crds/probe.go

```diff
@@ -0,0 +1,232 @@
+package kubernetes_crds
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/go-kit/log/level"
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	commonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+)
+
+func (cg *configGenerator) generateProbeConfig(m *v1.Probe) *config.ScrapeConfig {
+	c := config.DefaultScrapeConfig
+	cfg := &c
+	cfg.ScrapeInterval = config.DefaultGlobalConfig.ScrapeInterval
+	cfg.ScrapeTimeout = config.DefaultGlobalConfig.ScrapeTimeout
+	cfg.JobName = fmt.Sprintf("probe/%s/%s", m.Namespace, m.Name)
+	cfg.HonorTimestamps = true
+	cfg.MetricsPath = m.Spec.ProberSpec.Path
+	if m.Spec.Interval != "" {
+		cfg.ScrapeInterval, _ = model.ParseDuration(string(m.Spec.Interval))
+	}
+	if m.Spec.ScrapeTimeout != "" {
+		cfg.ScrapeInterval, _ = model.ParseDuration(string(m.Spec.ScrapeTimeout))
+	}
+	if m.Spec.ProberSpec.Scheme != "" {
+		cfg.Scheme = m.Spec.ProberSpec.Scheme
+	}
+	if m.Spec.ProberSpec.ProxyURL != "" {
+		if u, err := url.Parse(m.Spec.ProberSpec.ProxyURL); err != nil {
+			level.Error(cg.logger).Log("msg", "failed to parse ProxyURL from probe", "err", err)
+		} else {
+			cfg.HTTPClientConfig.ProxyURL = commonConfig.URL{URL: u}
+		}
+	}
+	return cfg
+}
+
+// 	if m.Spec.Module != "" {
```

Comment:
What's going on here? If we're going to leave commented code in like this, I'd like another comment explaining what it is and why it's being left in the file.